core-web/libs/edit-content/src/lib/utils/functions.util.spec.ts

```diff
@@ -0,0 +1,311 @@
+import { it, describe, expect } from '@jest/globals';
+
+import { DotCMSContentTypeField } from '@dotcms/dotcms-models';
+
+import * as functionsUtil from './functions.util';
+import { CALENDAR_FIELD_TYPES } from './mocks';
+
+import { FLATTENED_FIELD_TYPES } from '../models/dot-edit-content-field.constant';
+import { DotEditContentFieldSingleSelectableDataType } from '../models/dot-edit-content-field.enum';
+
+describe('Utils Functions', () => {
+    const { castSingleSelectableValue, getSingleSelectableFieldOptions, getFinalCastedValue } =
+        functionsUtil;
+
+    describe('castSingleSelectableValue', () => {
+        it('should return null if value is empty', () => {
+            expect(castSingleSelectableValue('', '')).toBeNull();
+        });
+
+        it('should return value if type is not Bool or Number', () => {
+            expect(castSingleSelectableValue('Some value', 'Some other type')).toBe('Some value');
+        });
+
+        describe('Boolean', () => {
+            it('should return true if value is true', () => {
+                expect(
+                    castSingleSelectableValue(
+                        'true',
+                        DotEditContentFieldSingleSelectableDataType.BOOL
+                    )
+                ).toBeTruthy();
```

Comment:
```suggestion
                ).toBe(true);
```

Reply:
This functions returns a boolean, so we can assert the booleat itself...

Reply:
What happen if I send `asedfsafasfd` or a number? but the second value is BOOL?

Reply:
it should return `false` in that case, I'll a test for that